test/integration/model.test.js

```diff
@@ -1698,10 +1698,81 @@ describe(Support.getTestDialectTeaser('Model'), () => {
 
       await this.User.bulkCreate(data);
       let affectedRows = await this.User.destroy({ where: { secretValue: '42' } });
-      expect(affectedRows).to.equal(2);
+      expect(affectedRows).to.be.a('number').to.equal(2);
       affectedRows = await this.User.destroy({ where: { secretValue: '44' } });
-      expect(affectedRows).to.equal(0);
+      expect(affectedRows).to.be.a('number').to.equal(0);
     });
+    if (dialect === 'postgres') {
+      const data = [
+        { username: 'Peter', secretValue: '42' },
+        { username: 'Paul', secretValue: '42' },
+        { username: 'Bob', secretValue: '43' },
+        { username: 'Rob', secretValue: '41' },
+        { username: 'Billy', secretValue: '40' },
+      ];
+      it('returns the affected rows if `options.returning` is set to true', async function () {
+        await this.User.bulkCreate(data);
+        let [affectedRowsCount, affectedRows] = await this.User.destroy({ where: { secretValue: '42' }, returning: true });
+        expect(affectedRowsCount).to.be.a('number');
+        expect(affectedRowsCount).to.equal(2);
+        expect(affectedRows).to.be.an('array').to.have.lengthOf(2);
+        [affectedRowsCount, affectedRows] = await this.User.destroy({ where: {}, returning: true });
+        expect(affectedRowsCount).to.be.a('number');
+        expect(affectedRowsCount).to.equal(3);
+        expect(affectedRows).to.be.an('array').to.have.lengthOf(3);
+      });
+
+      it('returns the affected rows count if `options.returning` is set to false', async function () {
+        await this.User.bulkCreate(data);
+        let affectedRowsCount = await this.User.destroy({ where: { secretValue: '42' }, returning: false });
+        expect(affectedRowsCount).to.be.a('number');
+        expect(affectedRowsCount).to.equal(2);
+        affectedRowsCount = await this.User.destroy({ where: {}, returning: false });
+        expect(affectedRowsCount).to.be.a('number');
+        expect(affectedRowsCount).to.equal(3);
+      });
+
+      it('returns only the fields in affected rows specified in `options.returning`', async function () {
+        await this.User.bulkCreate(data);
+        let [affectedRowsCount, affectedRows] = await this.User.destroy({ where: { secretValue: '42' }, returning: ['username'] });
+        expect(affectedRowsCount).to.be.a('number');
+        expect(affectedRowsCount).to.equal(2);
+        expect(affectedRows).to.be.an('array').to.have.lengthOf(2);
+        expect(affectedRows[0]).to.be.instanceOf(this.User);
+        JSON.parse(JSON.stringify(affectedRows)).forEach(userObj => {
```

Comment:
why serialize / deserialize this value? You could just check that the value is or isn't undefined on the model